### DISCUSSION TITLE
Novice Troubleshooting

### DISCUSSION THREAD

**Discussion by TH9705** [quartznet/quartznet#1584] [Q&A] [answered]
Running Quart.NET locally and working just fine. Just did my first deployment to a test environment and have hit a startup error and would appreciate some pointers for troubleshooting. Database user permissions for application account are set to `DELETE`, `INSERT`, `SELECT`, and `UPDATE`. I did adjust prefix to `[quartz].QRTZ_`, but that did not present a problem locally. Application stops running and logging after the last log message posted below.

```
"Registering datasource 'default' with db provider: 'Quartz.Impl.AdoJobStore.Common.DbProvider'","@i":"71509458","SourceContext":"Quartz.Util.DBConnectionManager"
"Using object serializer: Quartz.Simpl.JsonObjectSerializer, Quartz.Serialization.Json","@i":"9ba2a00e","SourceContext":"Quartz.Impl.StdSchedulerFactory"
"Initialized Scheduler Signaller of type: Quartz.Core.SchedulerSignalerImpl","@i":"d6ab0d5f","SourceContext":"Quartz.Core.SchedulerSignalerImpl"
"Quartz Scheduler created","@i":"389d80b0","SourceContext":"Quartz.Core.QuartzScheduler"
"JobFactory set to: Quartz.Simpl.MicrosoftDependencyInjectionJobFactory","@i":"17471663","SourceContext":"Quartz.Core.QuartzScheduler"
"Detected usage of SqlServerDelegate - defaulting 'selectWithLockSQL' to 'SELECT * FROM \"[quartz].QRTZ_\"LOCKS WITH (UPDLOCK,ROWLOCK) WHERE SCHED_NAME = @schedulerName AND LOCK_NAME = @lockName'.","@i":"d9881d4a","0":"[quartz].QRTZ_","SourceContext":"Quartz.Impl.AdoJobStore.JobStoreTX"
"Using db table-based data access locking (synchronization).","@i":"cb6a7186","SourceContext":"Quartz.Impl.AdoJobStore.JobStoreTX"
"JobStoreTX initialized.","@i":"73936189","SourceContext":"Quartz.Impl.AdoJobStore.JobStoreTX"
"Scheduler meta-data: Quartz Scheduler (v3.4.0.0) 'Startup' with instanceId 'abc-123-defghi637859206581450129'\n  Scheduler class: 'Quartz.Core.QuartzScheduler' - running locally.\n  NOT STARTED.\n  Currently in standby mode.\n  Number of jobs executed: 0\n  Using thread pool 'Quartz.Simpl.DefaultThreadPool' - with 5 threads.\n  Using job-store 'Quartz.Impl.AdoJobStore.JobStoreTX' - which supports persistence. and is clustered.\n","@i":"6b18eb76","SourceContext":"Quartz.Core.QuartzScheduler"
"Quartz scheduler 'Startup' initialized","@i":"77c58e3a","SourceContext":"Quartz.Impl.StdSchedulerFactory"
"Quartz scheduler version: 3.4.0.0","@i":"50ad033d","SourceContext":"Quartz.Impl.StdSchedulerFactory"
"Adding 1 jobs, 1 triggers.","@i":"3c519c58","SourceContext":"Quartz.ContainerConfigurationProcessor"
"ConnectionAndTransactionHolder passed to RollbackConnection was null, ignoring","@i":"7b25f01f","SourceContext":"Quartz.Impl.AdoJobStore.JobStoreTX"
```

Locally, the thing it would do next is the following, but I don't get this far.

```
"Prepared SQL: SELECT JOB_NAME,JOB_GROUP,DESCRIPTION,JOB_CLASS_NAME,IS_DURABLE,REQUESTS_RECOVERY,JOB_DATA FROM [quartz].QRTZ_JOB_DETAILS WHERE SCHED_NAME = @schedulerName AND JOB_NAME = @jobName AND JOB_GROUP = @jobGroup"
```

Any tips for tracking down this apparent database connection problem?

`Startup` class code, for reference:

```csharp
            services.AddQuartz(quartzConfigurator =>
            {
                quartzConfigurator.SchedulerId = "AUTO";
                quartzConfigurator.SchedulerName = nameof(Startup);
                quartzConfigurator.UseDefaultThreadPool(poolOptions => poolOptions
                    .MaxConcurrency = 5);
                quartzConfigurator.UseMicrosoftDependencyInjectionJobFactory();
                quartzConfigurator.UsePersistentStore(storeOptions =>
                {
                    storeOptions.UseSqlServer(adoProviderOptions =>
                    {
                        adoProviderOptions.ConnectionString = Configuration.GetConnectionString("ScrubbedName");
                        adoProviderOptions.TablePrefix = "[quartz].QRTZ_";
                    });
                    storeOptions.UseClustering();
                    storeOptions.UseJsonSerializer();
                    storeOptions.UseProperties = true;
                });
```

**Comment by jafin** [comment]
`SELECT * FROM \"[quartz].QRTZ_\"LOCKS`  looks broken based on where those quotes are (unsure why there are even quotes here but I'd have to dig into the source code) , are you using the same tableprefix locally and it works?

**Reply by TH9705** [reply]
Thank you for the prompt reply. Those quotes may be the difference between STDOUT logging in a Linux container vs. my Windows machine locally. Both cases are .NET 5  and the same source code, however. When I debug locally on my Windows machine the output appears without the quotes as:

`Detected usage of SqlServerDelegate - defaulting 'selectWithLockSQL' to 'SELECT * FROM [quartz].QRTZ_LOCKS WITH (UPDLOCK,ROWLOCK) WHERE SCHED_NAME = @schedulerName AND LOCK_NAME = @lockName'.`

Yes, table prefix is the same locally and working as expected. No separate configuration, just what I specify in that `Startup` snippet.

I think I may have a bad connection string and just a bad case of user error (mine). I will update this thread with my findings.

**Reply by lahma** [reply]
Maybe try to telnet/other TCP connection to SQL Server port or similar to ensure the connection is possible and not hanging for example?

**Comment by TH9705** [comment, marked as answer]
I have an answer and it was user error, unrelated to the Quartz library. I had a bad connection string when I deployed and should have caught my error with better logging elsewhere. Thanks to you both for your answers!
